doc/esmvalcore/datafinder.rst

```diff
@@ -1,7 +1,251 @@
+will allow the tool to search and find the right files. The specific
+datasets are listed in any recipe, under either the ``datasets`` and/or
+``additional_datasets`` sections, e.g. 
+
+.. code-block:: yaml
+
+  datasets:
+    - {dataset: HadGEM2-CC,  project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 2001, end_year: 2004}
+    - {dataset: UKESM1-0-LL, project: CMIP6, exp: historical, ensemble: r1i1p1f2, grid: gn, start_year: 2004,  end_year: 2014}
+
+``_data_finder`` will use this information to find data for **all** the variables specified in ``diagnostics/variables``.
+
+Recap and example
+=================
+Let us look at a practical example for a recap of the information above:
+suppose you are using a ``config-user.yml`` that has the following entries for
+data finding:
+
+.. code-block:: yaml
+
+  rootpath:  # running on CEDA-Jasmin
+    CMIP6: /badc/cmip6/data/CMIP6/CMIP
+  drs:
+    CMIP6: BADC  # since you are on CEDA-Jasmin
+
+and the dataset you need is specified in your ``recipe.yml`` as:
+
+.. code-block:: yaml
+
+  - {dataset: UKESM1-0-LL, project: CMIP6, mip: Amon, exp: historical, grid: gn, ensemble: r1i1p1f2, start_year: 2004,  end_year: 2014}
+
+for a variable, e.g.:
+
+.. code-block:: yaml
+
+  diagnostics:
+    some_diagnostic:
+      description: some_description
+      variables:
+        ta:
+          preprocessor: some_preprocessor
+
+The tool will then use the root path ``/badc/cmip6/data/CMIP6/CMIP`` and the
+dataset information and will assemble the full DRS path using information from
+CMOR-DRS_ and establish the path to the files as:
+
+``/badc/cmip6/data/CMIP6/CMIP/MOHC/UKESM1-0-LL/historical/r1i1p1f2/Amon``
+
+then look for variable ``ta`` and specifically the latest version of the data
+file: 
+
+``/badc/cmip6/data/CMIP6/CMIP/MOHC/UKESM1-0-LL/historical/r1i1p1f2/Amon/ta/gn/latest/``
+
+and finally, using the file naming definition from CMOR-DRS_ find the file:
+
+``/badc/cmip6/data/CMIP6/CMIP/MOHC/UKESM1-0-LL/historical/r1i1p1f2/Amon/ta/gn/latest/ta_Amon_UKESM1-0-LL_historical_r1i1p1f2_gn_195001-201412.nc``
+
+.. _observations:
+
+Observational data
+==================
+Observational data is retrieved in the same manner as CMIP data, for example
+using the ``OBS`` root path set to:
+
+  .. code-block:: yaml
+
+    OBS: /group_workspaces/jasmin4/esmeval/obsdata-v2
+
+and the dataset:
+
+  .. code-block:: yaml
+
+    - {dataset: ERA-Interim,  project: OBS,  type: reanaly,  version: 1,  start_year: 2014,  end_year: 2015,  tier: 3}
+
+in ``recipe.yml`` in ``datasets`` or ``additional_datasets``, the rules set in
+CMOR-DRS_ are used again and the file will be automatically found:
+
+``/group_workspaces/jasmin4/esmeval/obsdata-v2/Tier3/ERA-Interim/OBS_ERA-Interim_reanaly_1_Amon_ta_201401-201412.nc``
+
+Note that for observational data for ``drs: default`` the ``default`` directory
+must contain a sub-directory:
+``TierX`` (``Tier1``, ``Tier2`` or ``Tier3``).
```

Comment:
Do we need to explain here what the different "Tiers" stand for? Or is that explained somewhere else?

Reply:
It would be good to explain it (briefly) here, even if it does appear elsewhere. You never know the order that someone reads documentation - or how thoroughly they read it!

Reply:
Done.